src/commons/resources.rs

```diff
@@ -318,6 +319,107 @@ impl<T, K> Into<ResourceRequirements> for Resources<T, K> {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("missing {resource_key} resource {resource_policy} for container {container_name}")]
+pub struct ResourceRequirementsPolicyError {
```

Comment:
I struggle a bit with the word `Policy`, as it does not seem to be a k8s term (correct me if I'm wrong).
Finding an alternative is pretty hard. WDYT of simply calling it `ResourceRequirements` instead of `ResourceRequirementsPolicy` (through the PR)?

Reply:
The thing is that the term `ResourceRequirements` already exists (Imported from `k8s_openapi` and used across `operator-rs`. The term describes the complete struct, which contains the fields `requests`, `limits` and `claims`. One alternative name could be `ResourceRequirementsType`.

Reply:
Renamed in d48d06c